src/pitaya/src/ffi/metrics.rs

```diff
@@ -85,6 +94,25 @@ fn generic_register(
         CString::new(opts.subsystem.as_str()).expect("string should be valid inside rust");
     let name = CString::new(opts.name.as_str()).expect("string should be valid inside rust");
     let help = CString::new(opts.help.as_str()).expect("string should be valid inside rust");
+    let mut bucket_kind: CString = Default::default();
+    let buckets = opts
+        .buckets
+        .map(|b| {
+            bucket_kind =
```

Comment:
instead of assigning the variable inside the map, can't you return a tuple?
```rust
let (buckets, bucket_kind) = ...
```

Reply:
Well, it would have to return a tuple in the unwrap_or_else too with Default::default(), I don't know which one is more readable really.

Reply:
Yeah, this would be to just avoid the `mut`.

Reply:
I'm ok with both approaches and the change is really small so I see no problem with it.

Reply:
Done. PTAL?